tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/RotSharedImpl.java

```diff
@@ -0,0 +1,158 @@
+    // the first element.
+    StrictBitVector res = new StrictBitVector(
+        alist.get(0).getSize() + blist.get(0).getSize());
+    for (int i = 0; i < alist.size(); i++) {
+      // Multiply entry i of alist with entry i of blist
+      StrictBitVector temp = multiplyWithoutReduction(alist.get(i),
+          blist.get(i));
+      // XOR the product into the res holder as XOR constitutes multiplication
+      // in the binary Galois extension field.
+      res.xor(temp);
+    }
+    return res;
+  }
+
+  /**
+   * Computes the Galois product of two bit vectors, without reduction modulo a
+   * reduction polynomial.
+   *
+   * @param avec
+   *          The first bit vector
+   * @param bvec
+   *          The second bit vector
+   * @return The product represented as a StrictBitVector
+   */
+  private static StrictBitVector multiplyWithoutReduction(
+      StrictBitVector avec, StrictBitVector bvec) {
+    byte[] res = new byte[(avec.getSize() + bvec.getSize()) / 8];
+    byte[] avecBytes = avec.toByteArray();
+    byte[] bvecBytes = bvec.toByteArray();
+    for (int i = 0; i < avec.getSize(); i++) {
+      // If the i'th bit of avec is 1 then we shift the bvec i positions and xor
+      // the shifted bvec into the result vector, res
+      byte currentByte = (byte) (avecBytes[i / 8] >> (7 - (i % 8)));
+      if ((currentByte & 1) == 1) {
+        byte start = (byte) ((bvecBytes[0] << 24) >>> (24 + (i % 8)));
+        res[(i / 8) + 0] ^= start;
+        for (int j = 1; j < bvecBytes.length; j++) {
+          byte first = (byte) (bvecBytes[j - 1] << (8 - (i % 8)));
+          byte second = (byte) ((bvecBytes[j] << 24) >>> (24 + (i % 8)));
+          second ^= first;
+          res[(i / 8) + j] ^= second;
+        }
+        byte last = (byte) (bvecBytes[bvecBytes.length - 1] << (8 - (i % 8)));
+        res[(i / 8) + bvecBytes.length] ^= last;
+      }
+    }
+    return new StrictBitVector(res);
+  }
+
+  /**
+   * Compute a SHA-256 digest of elements in a list, concatenated with their
+   * index in the list. Only the first {@code size} elements of the list will be
+   * hashed.
+   *
+   * @param input
+   *          The list of StrictBitVector elements to hash. All elements MUST
+   *          have same length
+   * @param size
+   *          The amount of elements of the list, to hash. Must be less than or
+   *          equal to the amount of elements in the list.
+   * @return A list containing the hashed StrictBitVector as StrictBitVector
+   *         objects
+   */
+  protected List<StrictBitVector> hashBitVector(List<StrictBitVector> input, int size) {
+    List<StrictBitVector> res = new ArrayList<>(size);
+    // Allocate a buffer to contain the index of the value to hash along with
+    // the value itself.
+    // Size of an int is always 4 bytes in java
+    ByteBuffer indexBuffer = ByteBuffer
+        .allocate(4 + input.get(0).getSize() * 8);
```

Comment:
I think this should be `.allocate(4 + input.get(0).getSize() / 8);`

Reply:
... possibly even `.allocate((Integer.SIZE + input.get(0).getSize()) / 8);` to make the code more readable (and make the comment above redundant).

Reply:
Fixed in commit 0515efa5

Reply:
Actually, I think you missed some parenthesis. Nevermind, I will fix it before merge.